crates/engine/tree/src/tree/mod.rs

```diff
@@ -3664,6 +3775,28 @@ mod tests {
             assert_eq!(new[1].recovered_block().hash(), blocks[4].recovered_block().hash());
         }
 
+        // should be a None persistence action before we advance persistence
```

Comment:
nice! could we write/extend a test to cover the `ConsistentViewError::Reorged` error case?

Reply:
yep

Reply:
added two tests for the consistent view provider specifically

Reply:
awesome thanks!